pkg/util/ephemeral/interface.go

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ephemeral
+
+import "github.com/grafana/mimir/pkg/mimirpb"
+
+type SeriesCheckerByUser interface {
+	// EphemeralChecker returns an object which checks a series and decides whether
+	// this series should be ephemeral based on the given user's configuration.
+	// Can return nil if the user has no relevant configuration.
+	EphemeralChecker(string) SeriesChecker
+}
+
+type SeriesChecker interface {
+	// ShouldMarkEphemeral checks if a series with the given labelset should be marked as ephemeral.
+	ShouldMarkEphemeral(mimirpb.WriteRequest_SourceEnum, []mimirpb.LabelAdapter) bool
```

Comment:
Would it make sense to pass `source` to `SeriesCheckerByUser.EphemeralChecker`? Returned `SeriesChecker` would then be specific for that source.

It would also allow us to check if there are rules for given source, not "any rules" (in `HasMatchers()` method).

WDYT?

Reply:
hmm yeah, makes sense, that's more consistent with the fact that we already pass the `tenant` into `EphemeralChecker()`. I'll make this change.

Reply:
This is implemented now: https://github.com/grafana/mimir/pull/3897/files#diff-f85154121efb85199780877828359e09d6c77c2ae6bb94b601bd817b2c3798cd